Question-1.py

```diff
@@ -22,4 +22,60 @@
 # o Number of duplicate numbers: 2 # (2, 5 repeats in the list) 
 
 # If user select option E:
-# o List without duplicate numbers: [2, 3, 4, 12, 44, 1, 3]
\ No newline at end of file
+# o List without duplicate numbers: [2, 3, 4, 12, 44, 1, 3]
+
+
+
+# Solution:
+
+numbers = [2, 4, 5, 2, 12, 44, 5, 1, 2, 3]
+
+print("\n To select one of the following options just type 'A' or 'B' likewise....... \n")
+
+print(" A. Length of the list")
+
+print(" B. Display first 3 numbers")
+
+print(" C. Display sum of odd numbers")
+
+print(" D. Number of duplicate numbers")
+
+print(" E. Display list without duplicate numbers \n")
+
+choice = input("Type Here: ")
+
+print("You have choosen option",choice)
+
+# A. Length of the list
+if choice.lower() == 'a' :
+    print(f"Length of your list is {len(numbers)} ")
+
+# B. Display first 3 numbers
+elif choice.lower() == 'b':
+    print ("Your first three numbers are ", end=" ")
+    for x in numbers[:3]:
+        print(x, end=" ")
+
+# C. Display sum of odd numbers
+elif choice.lower() == 'c':
+    count=sum([i if i%2==1 else 0 for i in numbers])
+    print('Sum of all odd numbers is:',count )
+
+# D. Number of duplicate numbers  
+elif choice.lower() == 'd':
+    dups = []
+    count = 0
+    for num in numbers:
+        if num in dups:
+            count += 1
+        dups.append(num)
```

Comment:
The logic here fails. Read the doc properly.

For eg: [1, 2, 2, 3, 4, 2, 5, 6, 6, 7]

Here the output should be: 2 (Because the numbers that are repeating are 2 and 6 only)

But in your case it should give output like 3 for this input, which I think is incorrect. Please change the logic.

We don't want to count how many times the number is repeated, we want to count how many numbers are there that are repeating more than one in this case, they are 2 and 6,

Reply:
Use python collections.counter module:

You can simply count the number repeatations like:

[1, 2, 3, 2, 2, 4, 5, 5, 6]

{1: 1, 2: 3, 3:1, 4: 1, 5: 2, 6}

Here you can see only 2 and 5 repeated for more than one time then you can pick those keys and print it.

Reply:
Also please rename the `dict` to something else.

Reply:
i have updated the code with the use of counter as you suggested and renamed dict to something else.